FlashLFQ/Bayesian Protein Quantification/ProteinQuantificationEngineResult.cs

```diff
@@ -0,0 +1,115 @@
+        public readonly double NuPointEstimate;
+        public readonly (double, double) hdi_95;
+        public readonly List<double> foldChangeMeasurements;
+        public double PosteriorErrorProbability { get; private set; }
+        public double FalseDiscoveryRate { get; set; }
+        public double DegreeOfEvidence { get; private set; }
+        public double? cutoff { get; set; }
+
+        public ProteinQuantificationEngineResult(ProteinGroup protein, string condition1, string condition2, double[] mus, double[] sds, double[] nus,
+            List<double> fcs, double referenceIntensity)
+        {
+            this.protein = protein;
+            this.condition1 = condition1;
+            this.condition2 = condition2;
+            this.foldChangeMeasurements = fcs;
+            this.FoldChangePointEstimate = mus.Average();
+            this.StandardDeviationPointEstimate = sds.Average();
+            this.NuPointEstimate = nus.Average();
+            this.hdi_95 = Util.GetHighestDensityInterval(mus, 0.95);
+            this.ConditionIntensityPointEstimate = Math.Pow(2, FoldChangePointEstimate) * referenceIntensity;
+        }
+
+        /// <summary>
+        /// The posterior error probability (PEP) is the probability that the alternative hypothesis is not true.
+        /// 
+        /// This is estimated by dividing the number of Markov Chain Monte Carlo iterations that show a fold-change
+        /// less than the cutoff by the number of MCMC iterations that show a fold-change greater than the cutoff.
+        /// 
+        /// This means that if the null and alternative hypotheses are equally likely, the PEP equals 1.
+        /// If the null hypothesis is more likely than the alternative, the PEP is also equal to 1.
+        /// If the alternative hypothesis is more likely than the null hypothesis, then the PEP is less than 1.
+        /// As the alternative hypothesis becomes increasingly likely, the PEP decreases towards zero.
+        /// 
+        /// If there are no iterations of the MCMC algorithm that show a protein fold-change less than the cutoff,
+        /// then the PEP equals zero. 
+        /// 
+        /// If there are no iterations of the MCMC algorithm that show a protein fold-change greater than the absolute
+        /// value of the cutoff, then the PEP equals 1 and the null hypothesis can be accepted, rather than just
+        /// rejecting the alternative hypothesis.
+        /// 
+        /// Additional MCMC iterations can be performed to determine a more precise estimate of the PEP.
+        /// 
+        /// Numerically:
+        /// F = fold change
+        /// C = cutoff
+        /// 
+        /// if F > C, then P(H0|D) = P(F < C) / P (F > C)
+        /// if F < -C, then P(H0|D) = P(F > -C) / P (F < -C)
+        /// </summary>
+        public void CalculatePosteriorErrorProbability(double[] skepticalMus)
+        {
+            double pep = 1.0;
```

Comment:
What's the point of assigning a value to pep here and at ln 71 if you just overwrite it at ln 101? Delete these assignments.

Reply:
it's mostly to be robust against bugs. if there is some problem calculating the PEP, the protein will be marked as rejecting the alternative hypothesis, rather than accepting it. in other words, I want the default value of PEP to be 1.0 and not 0.0, which is the default assignment for `double`

Reply:
The statement "pep = nullHypothesisCount / alternativeHypothesisCount" will always produce a value that will overwrite the 1.0 or the value double.NaN. If there is a bug and one of those counts are null, an exception will be thrown that is not caught, which is not robust. If you suspect a bug could occur, use an exception and have the catch set the pep value to 1.0/NaN. The logic doesn't make sense as it currently reads.

Reply:
there can't be a crash because `double` can't be null but I see your point with the NaN, I'll fix that

Reply:
I'm still lost on how the pep = 1.0 would ever make it to the end of the method.

Reply:
> I see your point with the NaN, I'll fix that